libyang/data.py

```diff
@@ -224,6 +281,31 @@ def validate(
         if ret != 0:
             raise self.context.error("validation failed")
 
+    def diff(
+        self,
+        other: "DNode",
+        no_siblings: bool = False,
+        with_defaults: bool = False,
+    ) -> Iterator[DDiff]:
+        flags = diff_flags(no_siblings=no_siblings, with_defaults=with_defaults)
+        difflist = lib.lyd_diff(self.cdata, other.cdata, flags)
+
+        i = 0
+        while True:
```

Comment:
I'm not fond of `while True` constructs when it can be done differently.

Could you rework this like the following?

```python3
while difflist.type[i] != lib.LYD_DIFF_END:
```